crates/hyperqueue/src/dashboard/ui/screen.rs

```diff
@@ -1,10 +1,47 @@
+use tako::WorkerId;
 use termion::event::Key;
 
 use crate::dashboard::data::DashboardData;
 use crate::dashboard::ui::terminal::DashboardFrame;
 
 pub trait Screen {
     fn draw(&mut self, frame: &mut DashboardFrame);
-    fn update(&mut self, data: &DashboardData);
-    fn handle_key(&mut self, key: Key);
+    fn update(&mut self, data: &DashboardData, controller: &mut ScreenController);
+    fn handle_key(&mut self, key: Key, controller: &mut ScreenController);
+}
+
+#[derive(Clone, Copy)]
+pub enum DashboardScreenState {
+    ClusterOverviewScreen,
+    WorkerOverviewScreen(Option<WorkerId>),
+}
+
+pub struct ScreenController {
```

Comment:
Please turn `screen.rs` into a module and put the controller and the state into a separate file `controller.rs`.